File: 1967. number-of-strings-that-appear-as-substrings-in-word.java
class Solution {
    public int numOfStrings(String[] patterns, String word) {
        int count = 0;
        for (String s : patterns)
            if (word.contains(s)) count++;
        return count;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        System.out.println(sol.numOfStrings(
            new String[]{"a", "abc", "bc", "d"}, "abc"
        )); // Expected: 3

        System.out.println(sol.numOfStrings(
            new String[]{"a", "b", "c"}, "aaaaabbbbb"
        )); // Expected: 2

        System.out.println(sol.numOfStrings(
            new String[]{"a", "a", "a"}, "ab"
        )); // Expected: 3
    }
}

File: 1967. number-of-strings-that-appear-as-substrings-in-word.py
class Solution:
    def numOfStrings(self, patterns: list[str], word: str) -> int:
        count = 0
        for s in patterns:
            if s in word:
                count += 1
        return count


if __name__ == "__main__":
    sol = Solution()

    print(sol.numOfStrings(["a", "abc", "bc", "d"], "abc"))        # Expected: 3
    print(sol.numOfStrings(["a", "b", "c"], "aaaaabbbbb"))         # Expected: 2
    print(sol.numOfStrings(["a", "a", "a"], "ab"))                 # Expected: 3